pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -346,8 +358,8 @@ func maybeRetryDownload(ctx *downloaderContext,
 	status.RetryCount++
 	severity := types.GetErrorSeverity(status.RetryCount, time.Duration(status.RetryCount)*retryTime)
 	errDescription := types.ErrorDescription{
-		Error:               status.OrigError,
-		ErrorRetryCondition: fmt.Sprintf("Retrying; attempt %d", status.RetryCount),
+		Error:               status.Error,
+		ErrorRetryCondition: fmt.Sprintf("Retrying after %s; attempt %d", status.OrigError, status.RetryCount),
```

Comment:
Seems we should delete OrigError completely. It keeps first error and keep show it here. With deleting of this portion we will keep the latest error we have until receive new one or clear it. Also we check elapsed time since last error, seems setting an error again may affect this check.